### ISSUE TITLE
Adding drops and widgets to maps

### ISSUE THREAD

**Issue by cenveoanalyst** [pbugnion/gmaps#85]
Hello! I absolutely LOVE this package and it will be incredibly useful for making interactive maps. I am interested in adding drops and widgets to my maps. It appears by the source code there is a way to do that and the Google API obviously supports this. But there don't appear any examples in the documentation. How can I accomplish this?

**Comment by pbugnion** [comment]
Hi!  Glad you're finding this useful. 

What do you mean by "drops" and "widgets"? Do you mean having markers etc. (e.g. like [this](https://developers.google.com/maps/documentation/javascript/examples/marker-simple))? If so, I'm currently working on a PR for this (PR #78). No promises on when it will be ready but I may hopefully have a very basic version done by this weekend.

If you need something else, it would be great if you could post a link to relevant example in the [Google Maps JavaScript API](https://developers.google.com/maps/documentation/javascript/examples/). 

Thanks!

**Comment by cenveoanalyst** [comment]
Hello, that would certainly be a great step forward and I look forward to playing with it. I'm thinking that my end goal here would be to create [Fusion](https://developers.google.com/maps/documentation/javascript/examples/layer-fusiontables-simple) tables in the notebook. I want to create interactive maps from data sets I compute/provide. I'm soft on how Python and Javascript integrate together, but if you point me in the general direction I'll see what I can do to assist in the process of bringing this functionality to your library.

**Comment by pbugnion** [comment]
I've just had a play with fusion tables in Google maps -- it's very powerful. 

However, I'm still not entirely clear what you want. Would you like to interact with existing fusion tables? For instance, write something like:

```python
m = gmaps.Map()
bird_sightings_layer = gmaps.FusionTableLayer(query={'select': 'birds', 'from': 'table-id'})
m.add_layer(bird_sightings_layer)
m
```

... or do you mean you want functionality that can replace fusion tables, i.e. map very quickly from CSVs to maps (with lots of additional flexibility) in a jupyter notebook? The latter is ultimately what *gmaps* hopes to achieve, but we still have a long way to go.

For now, building a `FusionTableLayer` would certainly be feasible (and relatively straightforward) if you have a use case for it. I'd be glad to discuss interfaces etc. and give suggestions on how to get started.

**Comment by cenveoanalyst** [comment]
I guess the end goal is the second option. In looking through more of the documentation I think I've got a simpler goal in mind. I want to pull data into Pandas from our database, prepare it for import into Google by whatever means necessary, than plot exactly that data with Markers and Info Boxes prepared with whatever data I specify. 

So it could look something like this:

```
MarkerCoordinates = [(32.820441899999999, -79.993212),(41.9811306, -87.862998199999993)]
InfoBoxData = ['South Carolina DC', 'Rosemont IL DC']
markerlayer = gmaps.Marker(MarkerCoordinates)
infowindowlayer = gmaps.InfoWindowLayer(markers = MarkerCoordinates, data = InfoBoxData)
m = gmaps.Map()
m.add_layer(markerlayer)
m.add_layer(infowindowlayer)
m
```

So then the marker would be clickable and show the user the more detailed information. It would use [this](https://developers.google.com/maps/documentation/javascript/examples/infowindow-simple) or something like that. That seems a little simpler to implement as it could just piggy back off of what you're already doing. How can I get started with this? What are the interfaces involved?

**Comment by pbugnion** [comment]
This is definitely very achievable. 

Unfortunately, the documentation around how to implement jupyter widgets is fairly sparse. I suggest first having a look at the [ipywidgets documentation](http://ipywidgets.readthedocs.io/en/latest/). 

I think a good initial interface in Python could look like this:

```
marker_locations = [(10.0, 10.0), (20.0, 30.0)]
info_boxes = [gmaps.InfoBox('South Carolina'), gmaps.InfoBox('Vermont')]
markers = [
    gmaps.Marker(location=location, info_box=info_box)
    for location, info_box in zip(marker_locations, info_boxes)
]
marker_layer = gmaps.Markers(markers=markers)

m = gmaps.Map()
m.add_layer(marker_layer)
```

Basically, the info box layer becomes a property of the marker object itself. This will make it easier to bind a callback to the 'click' listener for the marker.

This could eventually be hidden behind some helper functions to create the marker layer, but we can worry about these later.

In terms of implementation, I suggest branching off the `feature/marker` branch. You then need to implement the following:

1. a view in JavaScript for the info box. Have a look at the `Marker` view for how to this. The view would really just create an instance of `google.maps.InfoWindow`. At this point, you can just fill the window with default text.

2. a listener on `click` in the marker (like [in the docs](https://developers.google.com/maps/documentation/javascript/examples/infowindow-simple)) which opens the window.

You should be able to get these working completely without touching the Python part. The next step is to create a model in Python. Have a look at the `Marker` model for an example of how to do this. The attributes of the model that get passed to the front-end are defined as [traitlets](https://github.com/ipython/traitlets). I think the only traitlet the `InfoBox` model would have would be `content`.

**Comment by pbugnion** [comment]
This is addressed in part by PR #74 , which is now merged and available in release 0.3.4. The major remaining part of this issue is the info box window.

**Comment by pbugnion** [comment]
The box window issue is addressed in PR #95 (thanks to bruno_messias).

**Comment by pbugnion** [comment]
Closed by PRs #74 and #98
